asciidoc.py

```diff
@@ -2265,7 +2300,11 @@ def gen_id(title):
         base_id = re.sub(r'\W+', '_', title).strip('_').lower()
         if 'ascii-ids' in document.attributes:
             # Replace non-ASCII characters with ASCII equivalents.
-            base_id = unicodedata.normalize('NFKD', base_id).encode('ascii', 'ignore').decode('ascii')
+            # Fixed implementation from https://stackoverflow.com/a/51230541 by Zohar Bar-Yehuda
+            normalized_id = ''.join(Section.NON_NFKD_MAP[c] if c in Section.NON_NFKD_MAP else c \
+                for part in unicodedata.normalize('NFKD', base_id) for c in part
+                if unicodedata.category(part) != 'Mn')
```

Comment:
I would prefer to not using nested generators with a filter. It's a confusing syntax at first glance, and prone to getting messed up in my experience in subsequent refactors by other people.

```suggestion
            base_id = ''
            for part in unicodedata.normalize('NFKD', base_id):
                if unicodedata.category(part) == 'Mn':
                    continue
                for character in part:
                    base_id += Section.NON_NFKD_MAP.get(character, character)
```